File: platform/sdk/python/examples/cloud/cloud-023_worm_legal_hold_on_off.py
import sys, os
sys.path.insert(0, os.path.join(os.path.dirname(__file__), "..", ".."))
from xuanji_sdk_cloud import CloudClient

client = CloudClient({"region": "cn-east-1"})
client.createBucket("lh-bucket")
client.putObject("lh-bucket", "hold.txt", "legal-hold-content")
client.wormLegalHoldOnOff("lh-bucket", "hold.txt", True)
result = client.wormLegalHoldOnOff("lh-bucket", "hold.txt", False)
if result["legalHold"] != "OFF":
    sys.exit(1)
print("XJ-OK: cloud-023_worm_legal_hold_on_off")
sys.exit(0)

File: platform/sdk/nodejs/examples/cloud/cloud-023_worm_legal_hold_on_off.js
const { CloudClient } = require("../../mox-sdk-cloud");
const client = new CloudClient({ region: "cn-east-1" });
client.createBucket("lh-bucket");
client.putObject("lh-bucket", "hold.txt", "legal-hold-content");
client.wormLegalHoldOnOff("lh-bucket", "hold.txt", true);
const result = client.wormLegalHoldOnOff("lh-bucket", "hold.txt", false);
if (result.legalHold !== "OFF") process.exit(1);
console.log("XJ-OK: cloud-023_worm_legal_hold_on_off");
process.exit(0);
